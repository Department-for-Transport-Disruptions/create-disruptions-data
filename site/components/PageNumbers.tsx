/* eslint-disable jsx-a11y/anchor-is-valid */
import { Dispatch, ReactElement, SetStateAction } from "react";

interface PageNumbersProps {
    currentPage: number;
    setCurrentPage: Dispatch<SetStateAction<number>>;
    numberOfPages: number;
}

const createPageNumbers = (
    numberOfPages: number,
    currentPage: number,
    setCurrentPage: Dispatch<SetStateAction<number>>,
) => {
    if (numberOfPages < 10) {
        const pageNumbers = [];

        for (let count = 0; count < numberOfPages; count++) {
            pageNumbers.push(
                <li
                    key={`page-${count + 1}`}
                    className={`govuk-pagination__item${
                        count + 1 === currentPage ? " govuk-pagination__item--current" : ""
                    }`}
                >
                    <a
                        onClick={(e) => {
                            e.preventDefault();
                            setCurrentPage(count + 1);
                        }}
                        className="govuk-link govuk-pagination__link"
                        href="#"
                        aria-label={`Page ${count + 1}`}
                    >
                        {count + 1}
                    </a>
                </li>,
            );
        }

        return pageNumbers;
    }

    if (currentPage === 1) {
        return [
            <li className="govuk-pagination__item govuk-pagination__item--current" key="page-1">
                <a
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(1);
                    }}
                    className="govuk-link govuk-pagination__link"
                    href="#"
                    aria-label="Page 1"
                >
                    1
                </a>
            </li>,
            <li className="govuk-pagination__item" key="page-2">
                <a
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(2);
                    }}
                    className="govuk-link govuk-pagination__link"
                    href="#"
                    aria-label="Page 2"
                >
                    2
                </a>
            </li>,
            <li className="govuk-pagination__item govuk-pagination__item--ellipses" key="ellipses-1">
                ...
            </li>,
            <li className="govuk-pagination__item" key={`page-${numberOfPages}`}>
                <a
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(numberOfPages);
                    }}
                    className="govuk-link govuk-pagination__link"
                    href="#"
                    aria-label={`Page ${numberOfPages}`}
                >
                    {numberOfPages}
                </a>
            </li>,
        ];
    }

    if (currentPage === numberOfPages) {
        return [
            <li className="govuk-pagination__item" key="page-1">
                <a
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(1);
                    }}
                    className="govuk-link govuk-pagination__link"
                    href="#"
                    aria-label="Page 1"
                >
                    1
                </a>
            </li>,
            <li className="govuk-pagination__item govuk-pagination__item--ellipses" key="ellipses-1">
                ...
            </li>,
            <li className="govuk-pagination__item" key={`page-${numberOfPages - 1}`}>
                <a
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(numberOfPages - 1);
                    }}
                    className="govuk-link govuk-pagination__link"
                    href="#"
                    aria-label={`Page ${numberOfPages - 1}`}
                >
                    {numberOfPages - 1}
                </a>
            </li>,
            <li className="govuk-pagination__item govuk-pagination__item--current" key={`page-${numberOfPages}`}>
                <a
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(numberOfPages);
                    }}
                    className="govuk-link govuk-pagination__link"
                    href="#"
                    aria-label={`Page ${numberOfPages}`}
                >
                    {numberOfPages}
                </a>
            </li>,
        ];
    }

    const pageNumbers = [
        currentPage - 1 !== 1 ? (
            <li className="govuk-pagination__item" key="page-1">
                <a
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(1);
                    }}
                    className="govuk-link govuk-pagination__link"
                    href="#"
                    aria-label="Page 1"
                >
                    1
                </a>
            </li>
        ) : (
            []
        ),
        currentPage !== 2 ? (
            <li className="govuk-pagination__item govuk-pagination__item--ellipses" key="ellipses-1">
                ...
            </li>
        ) : (
            []
        ),
        <li className="govuk-pagination__item" key={`page-${currentPage - 1}`}>
            <a
                onClick={(e) => {
                    e.preventDefault();
                    setCurrentPage(currentPage - 1);
                }}
                className="govuk-link govuk-pagination__link"
                href="#"
                aria-label={`Page ${currentPage - 1}`}
            >
                {currentPage - 1}
            </a>
        </li>,
        <li className="govuk-pagination__item govuk-pagination__item--current" key={`page-${currentPage}`}>
            <a
                onClick={(e) => {
                    e.preventDefault();
                    setCurrentPage(currentPage);
                }}
                className="govuk-link govuk-pagination__link"
                href="#"
                aria-label={`Page ${currentPage}`}
            >
                {currentPage}
            </a>
        </li>,
        <li className="govuk-pagination__item" key={`page-${currentPage + 1}`}>
            <a
                onClick={(e) => {
                    e.preventDefault();
                    setCurrentPage(currentPage + 1);
                }}
                className="govuk-link govuk-pagination__link"
                href="#"
                aria-label={`Page ${currentPage + 1}`}
            >
                {currentPage + 1}
            </a>
        </li>,
        currentPage !== numberOfPages - 1 ? (
            <li className="govuk-pagination__item govuk-pagination__item--ellipses" key="ellipses-2">
                ...
            </li>
        ) : (
            []
        ),
        currentPage + 1 !== numberOfPages ? (
            <li className="govuk-pagination__item" key={`page-${numberOfPages}`}>
                <a
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(numberOfPages);
                    }}
                    className="govuk-link govuk-pagination__link"
                    href="#"
                    aria-label={`Page ${numberOfPages}`}
                >
                    {numberOfPages}
                </a>
            </li>
        ) : (
            []
        ),
    ];

    return pageNumbers;
};

const PageNumbers = ({ currentPage, numberOfPages, setCurrentPage }: PageNumbersProps): ReactElement =>
    numberOfPages === 0 ? (
        <></>
    ) : (
        <nav className="govuk-pagination" role="navigation">
            {currentPage !== 1 ? (
                <div className="govuk-pagination__prev">
                    <a
                        onClick={(e) => {
                            e.preventDefault();
                            setCurrentPage(currentPage - 1);
                        }}
                        className="govuk-link govuk-pagination__link"
                        href=""
                        rel="prev"
                    >
                        <svg
                            className="govuk-pagination__icon govuk-pagination__icon--prev"
                            xmlns="http://www.w3.org/2000/svg"
                            height="13"
                            width="15"
                            aria-hidden="true"
                            focusable="false"
                            viewBox="0 0 15 13"
                        >
                            <path d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"></path>
                        </svg>
                        <span className="govuk-pagination__link-title">Previous</span>
                    </a>
                </div>
            ) : null}

            <ul className="govuk-pagination__list"> {createPageNumbers(numberOfPages, currentPage, setCurrentPage)}</ul>
            {currentPage !== numberOfPages ? (
                <div className="govuk-pagination__next">
                    <a
                        onClick={(e) => {
                            e.preventDefault();
                            setCurrentPage(currentPage + 1);
                        }}
                        className="govuk-link govuk-pagination__link"
                        href=""
                        rel="next"
                    >
                        <span className="govuk-pagination__link-title">Next</span>{" "}
                        <svg
                            className="govuk-pagination__icon govuk-pagination__icon--next"
                            xmlns="http://www.w3.org/2000/svg"
                            height="13"
                            width="15"
                            aria-hidden="true"
                            focusable="false"
                            viewBox="0 0 15 13"
                        >
                            <path d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"></path>
                        </svg>
                    </a>
                </div>
            ) : null}
        </nav>
    );
export default PageNumbers;
