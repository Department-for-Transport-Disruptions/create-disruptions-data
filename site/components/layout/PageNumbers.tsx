import Link from "next/link";
import { Dispatch, ReactElement, SetStateAction } from "react";

interface PageNumbersProps {
    currentPage: number;
    setCurrentPage: Dispatch<SetStateAction<number>>;
    numberOfPages: number;
}

const createPageNumbers = (
    numberOfPages: number,
    currentPage: number,
    setCurrentPage: Dispatch<SetStateAction<number>>,
) => {
    if (numberOfPages < 10) {
        const pageNumbers = [];

        for (let count = 0; count < numberOfPages; count++) {
            pageNumbers.push(
                <li
                    key={`page-${count + 1}`}
                    className={`govuk-pagination__item${
                        count + 1 === currentPage ? " govuk-pagination__item--current" : ""
                    }`}
                >
                    <Link
                        href="#"
                        role="button"
                        draggable="false"
                        className="govuk-link govuk-pagination__link"
                        data-module="govuk-button"
                        onClick={(e) => {
                            e.preventDefault();
                            setCurrentPage(count + 1);
                        }}
                        aria-label={`Page ${count + 1}`}
                    >
                        {count + 1}{" "}
                    </Link>
                </li>,
            );
        }

        return pageNumbers;
    }

    if (currentPage === 1) {
        return [
            <li className="govuk-pagination__item govuk-pagination__item--current" key="page-1">
                <Link
                    href="#"
                    role="button"
                    draggable="false"
                    className="govuk-link govuk-pagination__link"
                    data-module="govuk-button"
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(1);
                    }}
                    aria-label="Page 1"
                >
                    1
                </Link>
            </li>,
            <li className="govuk-pagination__item" key="page-2">
                <Link
                    href="#"
                    role="button"
                    draggable="false"
                    className="govuk-link govuk-pagination__link"
                    data-module="govuk-button"
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(2);
                    }}
                    aria-label="Page 2"
                >
                    2
                </Link>
            </li>,
            <li className="govuk-pagination__item govuk-pagination__item--ellipses" key="ellipses-1">
                ...
            </li>,
            <li className="govuk-pagination__item" key={`page-${numberOfPages}`}>
                <Link
                    href="#"
                    role="button"
                    draggable="false"
                    className="govuk-link govuk-pagination__link"
                    data-module="govuk-button"
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(numberOfPages);
                    }}
                    aria-label={`Page ${numberOfPages}`}
                >
                    {numberOfPages}
                </Link>
            </li>,
        ];
    }

    if (currentPage === numberOfPages) {
        return [
            <li className="govuk-pagination__item" key="page-1">
                <Link
                    href="#"
                    role="button"
                    draggable="false"
                    className="govuk-link govuk-pagination__link"
                    data-module="govuk-button"
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(1);
                    }}
                    aria-label="Page 1"
                >
                    1
                </Link>
            </li>,
            <li className="govuk-pagination__item govuk-pagination__item--ellipses" key="ellipses-1">
                ...
            </li>,
            <li className="govuk-pagination__item" key={`page-${numberOfPages - 1}`}>
                <Link
                    href="#"
                    role="button"
                    draggable="false"
                    className="govuk-link govuk-pagination__link"
                    data-module="govuk-button"
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(numberOfPages - 1);
                    }}
                    aria-label={`Page ${numberOfPages - 1}`}
                >
                    {numberOfPages - 1}
                </Link>
            </li>,
            <li className="govuk-pagination__item govuk-pagination__item--current" key={`page-${numberOfPages}`}>
                <Link
                    href="#"
                    role="button"
                    draggable="false"
                    className="govuk-link govuk-pagination__link"
                    data-module="govuk-button"
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(numberOfPages);
                    }}
                    aria-label={`Page ${numberOfPages}`}
                >
                    {numberOfPages}
                </Link>
            </li>,
        ];
    }

    const pageNumbers = [
        currentPage - 1 !== 1 ? (
            <li className="govuk-pagination__item" key="page-1">
                <Link
                    href="#"
                    role="button"
                    draggable="false"
                    className="govuk-link govuk-pagination__link"
                    data-module="govuk-button"
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(1);
                    }}
                    aria-label="Page 1"
                >
                    1
                </Link>
            </li>
        ) : (
            []
        ),
        currentPage !== 2 ? (
            <li className="govuk-pagination__item govuk-pagination__item--ellipses" key="ellipses-1">
                ...
            </li>
        ) : (
            []
        ),
        <li className="govuk-pagination__item" key={`page-${currentPage - 1}`}>
            <Link
                href="#"
                role="button"
                draggable="false"
                className="govuk-link govuk-pagination__link"
                data-module="govuk-button"
                onClick={(e) => {
                    e.preventDefault();
                    setCurrentPage(currentPage - 1);
                }}
                aria-label={`Page ${currentPage - 1}`}
            >
                {currentPage - 1}
            </Link>
        </li>,
        <li className="govuk-pagination__item govuk-pagination__item--current" key={`page-${currentPage}`}>
            <Link
                href="#"
                role="button"
                draggable="false"
                className="govuk-link govuk-pagination__link"
                data-module="govuk-button"
                onClick={(e) => {
                    e.preventDefault();
                    setCurrentPage(currentPage);
                }}
                aria-label={`Page ${currentPage}`}
            >
                {currentPage}
            </Link>
        </li>,
        <li className="govuk-pagination__item" key={`page-${currentPage + 1}`}>
            <Link
                href="#"
                role="button"
                draggable="false"
                className="govuk-link govuk-pagination__link"
                data-module="govuk-button"
                onClick={(e) => {
                    e.preventDefault();
                    setCurrentPage(currentPage + 1);
                }}
                aria-label={`Page ${currentPage + 1}`}
            >
                {currentPage + 1}
            </Link>
        </li>,
        currentPage !== numberOfPages - 1 ? (
            <li className="govuk-pagination__item govuk-pagination__item--ellipses" key="ellipses-2">
                ...
            </li>
        ) : (
            []
        ),
        currentPage + 1 !== numberOfPages ? (
            <li className="govuk-pagination__item" key={`page-${numberOfPages}`}>
                <Link
                    href="#"
                    role="button"
                    draggable="false"
                    className="govuk-link govuk-pagination__link"
                    data-module="govuk-button"
                    onClick={(e) => {
                        e.preventDefault();
                        setCurrentPage(numberOfPages);
                    }}
                    aria-label={`Page ${numberOfPages}`}
                >
                    {numberOfPages}
                </Link>
            </li>
        ) : (
            []
        ),
    ];

    return pageNumbers;
};

const PageNumbers = ({ currentPage, numberOfPages, setCurrentPage }: PageNumbersProps): ReactElement =>
    numberOfPages <= 1 ? (
        <></>
    ) : (
        <nav className="govuk-pagination justify-center">
            {currentPage !== 1 ? (
                <div className="govuk-pagination__prev">
                    <Link
                        href=""
                        role="button"
                        draggable="false"
                        className="govuk-link govuk-pagination__link"
                        data-module="govuk-button"
                        onClick={(e) => {
                            e.preventDefault();
                            setCurrentPage(currentPage - 1);
                        }}
                        aria-label={`Page ${numberOfPages}`}
                        rel="prev"
                    >
                        <svg
                            className="govuk-pagination__icon govuk-pagination__icon--prev"
                            xmlns="http://www.w3.org/2000/svg"
                            height="13"
                            width="15"
                            aria-hidden="true"
                            focusable="false"
                            viewBox="0 0 15 13"
                        >
                            <path d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z" />
                        </svg>
                        <span className="govuk-pagination__link-title">Previous</span>
                    </Link>
                </div>
            ) : null}

            <ul className="govuk-pagination__list"> {createPageNumbers(numberOfPages, currentPage, setCurrentPage)}</ul>
            {currentPage !== numberOfPages ? (
                <div className="govuk-pagination__next">
                    <Link
                        href=""
                        role="button"
                        draggable="false"
                        className="govuk-link govuk-pagination__link"
                        data-module="govuk-button"
                        onClick={(e) => {
                            e.preventDefault();
                            setCurrentPage(currentPage + 1);
                        }}
                        aria-label={`Page ${numberOfPages}`}
                        rel="next"
                    >
                        <span className="govuk-pagination__link-title">Next</span>{" "}
                        <svg
                            className="govuk-pagination__icon govuk-pagination__icon--next"
                            xmlns="http://www.w3.org/2000/svg"
                            height="13"
                            width="15"
                            aria-hidden="true"
                            focusable="false"
                            viewBox="0 0 15 13"
                        >
                            <path d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z" />
                        </svg>
                    </Link>
                </div>
            ) : null}
        </nav>
    );
export default PageNumbers;
