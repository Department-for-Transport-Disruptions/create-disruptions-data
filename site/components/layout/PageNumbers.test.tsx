import { render } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import PageNumbers from "./PageNumbers";

describe("PageNumbers", () => {
    it("should not render anything when there are no pages", () => {
        const { asFragment } = render(<PageNumbers currentPage={0} numberOfPages={0} setCurrentPage={vi.fn()} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should not render anything when there is one page", () => {
        const { asFragment } = render(<PageNumbers currentPage={1} numberOfPages={1} setCurrentPage={vi.fn()} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render correctly when there are less than 10 pages", () => {
        const { asFragment } = render(<PageNumbers currentPage={1} numberOfPages={8} setCurrentPage={vi.fn()} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render correctly when there are 10 pages and the current page is 1", () => {
        const { asFragment } = render(<PageNumbers currentPage={1} numberOfPages={10} setCurrentPage={vi.fn()} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it("should render correctly when there are 10 pages and the current page is 5", () => {
        const { asFragment } = render(<PageNumbers currentPage={5} numberOfPages={10} setCurrentPage={vi.fn()} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
