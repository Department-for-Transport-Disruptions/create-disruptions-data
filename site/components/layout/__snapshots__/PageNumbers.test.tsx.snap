// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageNumbers > should not render anything when there are no pages 1`] = `<DocumentFragment />`;

exports[`PageNumbers > should not render anything when there is one page 1`] = `<DocumentFragment />`;

exports[`PageNumbers > should render correctly when there are 10 pages and the current page is 1 1`] = `
<DocumentFragment>
  <nav
    class="govuk-pagination justify-center"
  >
    <ul
      class="govuk-pagination__list"
    >
       
      <li
        class="govuk-pagination__item govuk-pagination__item--current"
      >
        <a
          aria-label="Page 1"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          1
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 2"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          2
        </a>
      </li>
      <li
        class="govuk-pagination__item govuk-pagination__item--ellipses"
      >
        ...
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 10"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          10
        </a>
      </li>
    </ul>
    <div
      class="govuk-pagination__next"
    >
      <a
        aria-label="Page 10"
        class="govuk-link govuk-pagination__link"
        data-module="govuk-button"
        draggable="false"
        href=""
        rel="next"
        role="button"
      >
        <span
          class="govuk-pagination__link-title"
        >
          Next
        </span>
         
        <svg
          aria-hidden="true"
          class="govuk-pagination__icon govuk-pagination__icon--next"
          focusable="false"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"
          />
        </svg>
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`PageNumbers > should render correctly when there are 10 pages and the current page is 5 1`] = `
<DocumentFragment>
  <nav
    class="govuk-pagination justify-center"
  >
    <div
      class="govuk-pagination__prev"
    >
      <a
        aria-label="Page 10"
        class="govuk-link govuk-pagination__link"
        data-module="govuk-button"
        draggable="false"
        href=""
        rel="prev"
        role="button"
      >
        <svg
          aria-hidden="true"
          class="govuk-pagination__icon govuk-pagination__icon--prev"
          focusable="false"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m6.5938-0.0078125-6.7266 6.7266 6.7441 6.4062 1.377-1.449-4.1856-3.9768h12.896v-2h-12.984l4.2931-4.293-1.414-1.414z"
          />
        </svg>
        <span
          class="govuk-pagination__link-title"
        >
          Previous
        </span>
      </a>
    </div>
    <ul
      class="govuk-pagination__list"
    >
       
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 1"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          1
        </a>
      </li>
      <li
        class="govuk-pagination__item govuk-pagination__item--ellipses"
      >
        ...
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 4"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          4
        </a>
      </li>
      <li
        class="govuk-pagination__item govuk-pagination__item--current"
      >
        <a
          aria-label="Page 5"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          5
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 6"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          6
        </a>
      </li>
      <li
        class="govuk-pagination__item govuk-pagination__item--ellipses"
      >
        ...
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 10"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          10
        </a>
      </li>
    </ul>
    <div
      class="govuk-pagination__next"
    >
      <a
        aria-label="Page 10"
        class="govuk-link govuk-pagination__link"
        data-module="govuk-button"
        draggable="false"
        href=""
        rel="next"
        role="button"
      >
        <span
          class="govuk-pagination__link-title"
        >
          Next
        </span>
         
        <svg
          aria-hidden="true"
          class="govuk-pagination__icon govuk-pagination__icon--next"
          focusable="false"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"
          />
        </svg>
      </a>
    </div>
  </nav>
</DocumentFragment>
`;

exports[`PageNumbers > should render correctly when there are less than 10 pages 1`] = `
<DocumentFragment>
  <nav
    class="govuk-pagination justify-center"
  >
    <ul
      class="govuk-pagination__list"
    >
       
      <li
        class="govuk-pagination__item govuk-pagination__item--current"
      >
        <a
          aria-label="Page 1"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          1 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 2"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          2 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 3"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          3 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 4"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          4 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 5"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          5 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 6"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          6 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 7"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          7 
        </a>
      </li>
      <li
        class="govuk-pagination__item"
      >
        <a
          aria-label="Page 8"
          class="govuk-link govuk-pagination__link"
          data-module="govuk-button"
          draggable="false"
          href="#"
          role="button"
        >
          8 
        </a>
      </li>
    </ul>
    <div
      class="govuk-pagination__next"
    >
      <a
        aria-label="Page 8"
        class="govuk-link govuk-pagination__link"
        data-module="govuk-button"
        draggable="false"
        href=""
        rel="next"
        role="button"
      >
        <span
          class="govuk-pagination__link-title"
        >
          Next
        </span>
         
        <svg
          aria-hidden="true"
          class="govuk-pagination__icon govuk-pagination__icon--next"
          focusable="false"
          height="13"
          viewBox="0 0 15 13"
          width="15"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m8.107-0.0078125-1.4136 1.414 4.2926 4.293h-12.986v2h12.896l-4.1855 3.9766 1.377 1.4492 6.7441-6.4062-6.7246-6.7266z"
          />
        </svg>
      </a>
    </div>
  </nav>
</DocumentFragment>
`;
